src/main/kotlin/org/rust/cargo/toolchain/impl/CargoMetadata.kt

```diff
@@ -170,9 +171,29 @@ object CargoMetadata {
         val id: PackageId,
 
         /**
-         * id's of dependent packages
+         * Id's of dependent packages
          */
-        val dependencies: List<PackageId>
+        val dependencies: List<PackageId>,
+
+        /**
+         * List of dependency info
+         *
+         * Contains additional info compared with [dependencies] like custom package name.
+         * Can be null for old cargo version
+         */
+        val deps: List<Dep>?
+    )
+
+    data class Dep(
+        /**
+         * Id of dependent package
+         */
+        val pkg: PackageId,
+
+        /**
+         * Dependency name that should be used in code as extern crate name
+         */
+        val name: String?
```

Comment:
I think this is guaranteed to be non-null

Reply:
Hm, [original structure](https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_output_metadata.rs#L107-L110) in cargo also holds dependency name as `Option<String>`

Reply:
Oh, you are right... The reason for that is horrible: it's possible to depend on a carte which contains only binary target. So, while dependency is present, it is sort-of a no-op dependency.